themes/perun/perun/disco-tpl.php

```diff
@@ -23,165 +24,149 @@
 $this->data['head'] .= '<link rel="stylesheet" media="screen" type="text/css" href="' .
     Module::getModuleUrl('perun/res/css/disco.css') . '" />';
 
-$this->data['head'] .= '<script type="text/javascript" src="' .
-    Module::getModuleUrl('perun/res/js/jquery.livesearch.js') . '"></script>';
-
-$this->data['head'] .= '<script type="text/javascript" src="' .
-    Module::getModuleUrl('discopower/assets/js/suggest.js') . '"></script>';
-
-$this->data['head'] .= Disco::searchScript();
-$this->data['head'] .= Disco::showEntriesScript();
-$this->data['head'] .= Disco::setFocus();
-
-const CONFIG_FILE_NAME = 'module_perun.php';
-
-const ADD_INSTITUTION_URL = 'disco.addInstitution.URL';
-const ADD_INSTITUTION_EMAIL = 'disco.addInstitution.email';
-
-const URN_CESNET_PROXYIDP_IDPENTITYID = 'urn:cesnet:proxyidp:idpentityid:';
-
-const WARNING_TYPE_INFO = 'INFO';
-const WARNING_TYPE_WARNING = 'WARNING';
-const WARNING_TYPE_ERROR = 'ERROR';
-
-$warningIsOn = $this->data['warningIsOn'];
-$warningType = $this->data['warningType'];
-$warningTitle = $this->data['warningTitle'];
-$warningText = $this->data['warningText'];
-
+$warningAttributes = $this->data[Disco::WARNING_ATTRIBUTES];
+$this->includeInlineTranslation('{perun:disco:warning_title}', $warningAttributes->getTitle());
+$this->includeInlineTranslation('{perun:disco:warning_text}', $warningAttributes->getText());
 $authContextClassRef = null;
 $idpEntityId = null;
 
 $config = null;
 
 $addInstitutionUrl = '';
 $addInstitutionEmail = '';
+$translate_module = '';
+$wayfConfig = [];
 
+//LOAD CONFIG
 try {
-    $config = Configuration::getConfig(CONFIG_FILE_NAME);
+    $config = Configuration::getConfig(Disco::CONFIG_FILE_NAME);
 } catch (\Exception $ex) {
-    Logger::warning('perun:disco-tpl: missing or invalid module_perun.php config file');
+    Logger::error('perun:disco-tpl: missing or invalid module_perun.php config file');
+    throw $ex;
 }
 
 if ($config !== null) {
     try {
-        $addInstitutionUrl = $config->getString(ADD_INSTITUTION_URL);
+        $wayfConfig = $config->getConfigItem(Disco::WAYF);
     } catch (\Exception $ex) {
-        Logger::warning('perun:disco-tpl: missing or invalid addInstitution.URL parameter in module_perun.php file');
+        Logger::error("perun:disco-tpl: missing configuration for param '" . Disco::WAYF . "'");
+        throw $ex;
+    }
+    $translate_module = $wayfConfig->getString(Disco::TRANSLATE_MODULE, 'disco');
+    $addInstitution = $wayfConfig->getConfigItem(Disco::ADD_INSTITUTION);
+    try {
+        $addInstitutionUrl = $addInstitution->getString(Disco::ADD_INSTITUTION_URL);
```

Comment:
I think that will be better to use some prefix/suffix for configuration blocks and you use more case style in one code 

'transtlate_module' vs 'addInstitution'

Reply:
With the prefix/suffix, do you mean to have something like
```php
'add_instiution' => [
    'add_inst_email' => '...',
    'add_inst_url' => '...',
]
```
instead of 
```php
'add_instiution' => [
    'email' => '...',
    'url' => '...',
]
```
?
The case should be the same in all properties, I will however double check.